src/networking.c

```diff
@@ -565,7 +658,17 @@ cc_oci_network_discover(struct cc_oci_config *const config,
 			ipv4_cfg->subnet_mask = cc_net_get_ip_address(family,
 				&((struct sockaddr_in *)ifa->ifa_netmask)->sin_addr);
 
-
+			/* Retrieve the mtu set by docker for the interface to be set
+			 * later for the tap interface sent to the VM and 
+			 * the interface within the VM
+			 */
+			if (if_cfg->mtu == 0) {
+				if ( cc_oci_get_interface_mtu(if_cfg->ifname, &mtu)) {
+					if_cfg->mtu = mtu;
+				} else {
+					return false;
```

Comment:
This code leaks resources:

- there is no call to `freeifaddrs()`
- `if_cfg` might need freeing.